### PR TITLE
webhook signature

### PR DESCRIPTION
**Pull request by shimonewman**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: app/workers/webhooks.php [hunk 1]
```diff
@@ -68,7 +70,7 @@ protected function execute(array $events, string $payload, Document $webhook, Do
                 'X-' . APP_NAME . '-Webhook-Name: ' . $webhook->getAttribute('name', ''),
                 'X-' . APP_NAME . '-Webhook-User-Id: ' . $user->getId(),
                 'X-' . APP_NAME . '-Webhook-Project-Id: ' . $project->getId(),
-                'X-' . APP_NAME . '-Webhook-Signature: ' . $webhook->getAttribute('signature', 'not-yet-implemented'),
+                'X-' . APP_NAME . '-Webhook-Signature: ' . $webhook->getAttribute('signature', $signature),
```
Comment (TorstenDittmann) [inline app/workers/webhooks.php:73]:
```suggestion
                'X-' . APP_NAME . '-Webhook-Signature: ' . $signature,
```

We don't wanna ask the DB here 👍🏻

File: src/Appwrite/Utopia/Response/Model/Webhook.php [hunk 2]
```diff
@@ -58,6 +58,12 @@ public function __construct()
                 'default' => '',
                 'example' => 'password',
             ])
+            ->addRule('signatureKey', [
+                'type' => self::TYPE_STRING,
+                'description' => 'random string to be used as key to encrypt signature',
```
Comment (TorstenDittmann) [inline src/Appwrite/Utopia/Response/Model/Webhook.php:63]:
```suggestion
                'description' => 'Signature key which can be used to validated incoming Webhooks.',
```

File: app/controllers/api/projects.php [hunk 3]
```diff
@@ -714,6 +718,7 @@
             ->setAttribute('security', $security)
             ->setAttribute('httpUser', $httpUser)
             ->setAttribute('httpPass', $httpPass)
+            ->setAttribute('signatureKey', empty($signatureKey) ? \bin2hex(\random_bytes(64)) : $signatureKey)
```
[the remark this answers is not in the document]
Comment (shimonewman) [reply app/controllers/api/projects.php:721]: Cool

File: app/workers/webhooks.php (hunk 1 above)
Comment (shimonewman) [reply app/workers/webhooks.php:73]: Opsss

File: tests/e2e/Services/Webhooks/WebhooksBase.php [hunk 4]
```diff
@@ -30,6 +30,10 @@ public function testCreateCollection(): array
         $this->assertNotEmpty($actors['body']['$id']);
 
         $webhook = $this->getLastRequest();
+        $signatureKey = $this->getProject()['signatureKey'];
+        $payload = json_encode($webhook['data']);
+        $url     = $webhook['url'];
+        $signatureExpected = base64_encode(hash_hmac('sha1', $url . $payload, $signatureKey, true));
```
[the remark this answers is not in the document]
Comment (shimonewman) [reply tests/e2e/Services/Webhooks/WebhooksBase.php:36]: Sure

File: src/Appwrite/Utopia/Response/Model/Webhook.php (hunk 2 above)
Comment (shimonewman) [reply src/Appwrite/Utopia/Response/Model/Webhook.php:63]: Cool

**Review by TorstenDittmann — COMMENTED** [review]
Awesome, just 2 tiny comments and I think that's it.

To fix the linter tests, you can run the following:
```sh
./vendor/bin/phpcbf
```
This will run the formatter according to our rules 👍🏻 

Reference: https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md#code-maintenance

**Pull request merged**